### ISSUE TITLE
Issue When the Writer is Closed via Timeout

### ISSUE THREAD

**Issue by ghaffarian** [sahabpardaz/kafka-parquet-writer#12]
I encountered an issue when the writer is interrupted to finalize a file. After the maximum open-duration of a file expires, the program starts printing the following logs at a very high rate:
```
...
2019-11-04 15:50:59,171 INFO  KafkaProtoParquetWriter:346 [KafkaProtoParquetWriter-test-instance-0] - Acking the records on Kafka consumer interrupted.
2019-11-04 15:50:59,171 INFO  KafkaProtoParquetWriter:346 [KafkaProtoParquetWriter-test-instance-0] - Acking the records on Kafka consumer interrupted.
2019-11-04 15:50:59,171 INFO  KafkaProtoParquetWriter:346 [KafkaProtoParquetWriter-test-instance-0] - Acking the records on Kafka consumer interrupted.
2019-11-04 15:50:59,171 INFO  KafkaProtoParquetWriter:346 [KafkaProtoParquetWriter-test-instance-0] - Acking the records on Kafka consumer interrupted.
2019-11-04 15:50:59,182 INFO  KafkaProtoParquetWriter:346 [KafkaProtoParquetWriter-test-instance-0] - Acking the records on Kafka consumer interrupted.
2019-11-04 15:50:59,182 INFO  KafkaProtoParquetWriter:346 [KafkaProtoParquetWriter-test-instance-0] - Acking the records on Kafka consumer interrupted.
2019-11-04 15:50:59,182 INFO  KafkaProtoParquetWriter:346 [KafkaProtoParquetWriter-test-instance-0] - Acking the records on Kafka consumer interrupted.
2019-11-04 15:50:59,182 INFO  KafkaProtoParquetWriter:346 [KafkaProtoParquetWriter-test-instance-0] - Acking the records on Kafka consumer interrupted.
...
```

As you can see the log is generated at line 346 of `KafkaProtoParquetWriter.java`. Taking a look at the code, it seems that the program is failing to do `smartCommitKafkaConsumer.ack(...)` at line 344 with an InterruptedException.

Isn't it better to not catch the exception and just let it get thrown out?

**Comment by ghaffarian** [comment]
After doing some more analysis, it also seems to be related to the order of closing the `worker-threads` and `smartCommitKafkaConsumer` in the `close()` method of `KafkaProtoParquetWriter` (lines: 188 and 190).

**Comment by bozorgzadeh** [comment]
It is solved in version 1.4.0
